ingester-prometheus-metrics/src/main/java/io/greptime/metrics/MetricsExporter.java

```diff
@@ -27,29 +27,30 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class MetricsExporter implements Lifecycle<Void> {
+public class MetricsExporter implements Lifecycle<ExporterOptions> {
 
     private static final Logger LOG = LoggerFactory.getLogger(MetricsExporter.class);
 
     private final CollectorRegistry prometheusMetricRegistry;
 
-    private final int port;
     private HTTPServer server;
+    private ExporterOptions opts;
 
     private final AtomicBoolean started = new AtomicBoolean(false);
 
-    public MetricsExporter(int port, MetricRegistry dropwizardMetricRegistry) {
-        this.port = port;
+    public MetricsExporter(MetricRegistry dropwizardMetricRegistry) {
         this.prometheusMetricRegistry = new CollectorRegistry();
         this.prometheusMetricRegistry.register(new DropwizardExports(dropwizardMetricRegistry));
     }
 
     @Override
-    public boolean init(Void opts) {
+    public boolean init(ExporterOptions opts) {
         if (this.started.compareAndSet(false, true)) {
+            this.opts = opts;
             try {
-                this.server = new HTTPServer(new InetSocketAddress(this.port), this.prometheusMetricRegistry, true);
-                LOG.info("Metrics exporter started at `http://localhost:{}/metrics`", this.port);
+                this.server = new HTTPServer(
+                        new InetSocketAddress(opts.getPort()), this.prometheusMetricRegistry, opts.isDeamon());
```

Comment:
Does `InetSocketAddress` bind to "0.0.0.0" by default? If not, better set it explicitly, useful in cloud or docker environment.